src/test/java/io/confluent/connect/jdbc/sink/integration/PostgresDatatypeIT.java

```diff
@@ -42,67 +47,167 @@
   private String tableName;
-  private JdbcSinkTask task;
+  private JsonConverter jsonConverter;
+  private Map<String, String> props;
 
   @Before
   public void before() {
+    startConnect();
+    jsonConverter = jsonConverter();
+    props = baseSinkProps();
+
     tableName = "test";
-    props = new HashMap<>();
     String jdbcURL = String
         .format("jdbc:postgresql://localhost:%s/postgres", pg.getEmbeddedPostgres().getPort());
     props.put(JdbcSinkConfig.CONNECTION_URL, jdbcURL);
     props.put(JdbcSinkConfig.CONNECTION_USER, "postgres");
     props.put("pk.mode", "none");
     props.put("topics", tableName);
+
+    // create topic in Kafka
+    connect.kafka().createTopic(tableName, 1);
   }
 
   @After
   public void after() throws SQLException {
-    stopTask();
     try (Connection c = pg.getEmbeddedPostgres().getPostgresDatabase().getConnection()) {
       try (Statement s = c.createStatement()) {
         s.execute("DROP TABLE IF EXISTS " + tableName);
       }
     }
     log.info("Dropped table");
+    stopConnect();
+  }
+
+  /**
+   * Verifies that even when the connector encounters exceptions that would cause a connection
+   * with an invalid transaction, the connector sends only the errant record to the error
+   * reporter and establishes a valid transaction for subsequent correct records to be sent to
+   * the actual database.
+   */
+  @Test
+  public void testPrimaryKeyConstraintsSendsToErrorReporter() throws Exception {
+    props.put(ERRORS_TOLERANCE_CONFIG, ToleranceType.ALL.value());
+    props.put(DLQ_TOPIC_NAME_CONFIG, DLQ_TOPIC_NAME);
+    props.put(DLQ_TOPIC_REPLICATION_FACTOR_CONFIG, "1");
+    props.put(MAX_RETRIES, "2");
+
+    createTableWithPrimaryKey();
+    connect.configureConnector("jdbc-sink-connector", props);
+    waitForConnectorToStart("jdbc-sink-connector", 1);
+
+    final Schema schema = SchemaBuilder.struct().name("com.example.Person")
+        .field("firstname", Schema.STRING_SCHEMA)
+        .field("lastname", Schema.STRING_SCHEMA)
+        .build();
+    final Struct firstStruct = new Struct(schema)
+        .put("firstname", "Christina")
+        .put("lastname", "Brams");
+
+    produceRecord(schema, firstStruct);
+    // Send the same record for a PK collision
+    produceRecord(schema, firstStruct);
+
+    // Now, create and send another normal record
+    Struct secondStruct = new Struct(schema)
+        .put("firstname", "Brams")
+        .put("lastname", "Christina");
+
+    produceRecord(schema, secondStruct);
+
+    waitForCommittedRecords("jdbc-sink-connector", Collections.singleton(tableName), 3, 1,
+        TimeUnit.MINUTES.toMillis(3));
+
+    KafkaConsumer<byte[], byte[]> consumer =
+        connect.kafka().createConsumerAndSubscribeTo(Collections.emptyMap(), DLQ_TOPIC_NAME);
+
+    ConsumerRecords<byte[], byte[]> records =
+        consumer.poll(Duration.ofMillis(CONSUME_MAX_DURATION_MS));
+
+    assertEquals(1, records.count());
```

Comment:
Why are we manually constructing a consumer here instead of doing something like `connect.kafka().consume(1, CONSUME_MAX_DURATION_MS, DLQ_TOPIC_NAME)`?

Reply:
Fixed this.